researchspace/researchspace
# Added custom dropdown to Knowledge Map and Semantic Narrative toolbar

# What

Semantic Narrative
- updated toolbar to show a custom dropdown menu for any existing semantic narrative
- added new events
  - `Narrative.Created` triggered when a new narrative is created
  - `Narrative.Updated` triggered when an existing narrative is updated  

Knowledge Map toolbar
- updated toolbar to show a custom dropdown menu for any knowledge map

# Video / Gif / Screenshot

### Semantic Narrative

#### New narrative - dropdown not visible

![Screenshot 2024-07-22 at 16 28 42](https://github.com/user-attachments/assets/1cfbca82-2c2c-467b-a055-7d9435f2ffd7)

#### Existing narrative - dropdown visible

![Screenshot 2024-07-22 at 16 28 55](https://github.com/user-attachments/assets/2e9790b3-c5af-4e37-8889-d66e2b703c27)


### Knowledge Map

#### New Knowledge Map - dropdown not visible

![Screenshot 2024-07-22 at 16 29 24](https://github.com/user-attachments/assets/b88bc92f-c6f6-45e2-a778-50a918cc1c15)


#### Existing Knowledge Map - dropdown visible
![Screenshot 2024-07-22 at 16 29 37](https://github.com/user-attachments/assets/36adb52f-e665-40b7-a65b-edd1c493669d)

# Meta

Updated some formatting style

# How To Test

Semantic Narrative
- Go to `Semantic Narrative` and click to `New Semantic Narrative` --> it should open a new semantic narrative view with the toolbar without the custom dropdown
- Open an existing `Semantic Narrative` --> the custom dropdown should be visible in the toolbar

Knowledge Map
- Go to `Knowledge Map` and click to `New Knowledge Map` --> it should open a new knowledge map view with the toolbar without the custom dropdown
- Open an existing `Knowledge Map` --> the custom dropdown should be visible in the toolbar
